### PR TITLE
LPS-136259 Avoid years rounding by Moment.js library

### PR DESCRIPTION
**Pull request by alinedoleron**
Ticket: https://issues.liferay.com/browse/LPS-136259

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -228,15 +230,34 @@ const DatePicker = ({
...
+				else if (year <= 50) {
+					inputRef.current.value = moment(initialValueMemoized)
+						.subtract(2000, 'years')
+						.format(dateMask.toUpperCase());
+				}
+				else {
+					inputRef.current.value = moment(
+						initialValueMemoized
+					).format(dateMask.toUpperCase());
+				}
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:252]: I suggest you to change the order the `if`s occur and the way you  set the `inputRef` to simplify the code and increase the code reuse:
```suggestion
				const date = moment(initialValueMemoized);
				
				if (year <= 50) {
					date.subtract(1900, 'years');
				}
				else if (year < 100) {
					date.subtract(2000, 'years');
				}
				
				inputRef.current.value = date.format(dateMask.toUpperCase());
```

**Review by miltonmc — CHANGES_REQUESTED** [review]
Good job!

Just a suggestion to simplify things:

**Comment by alinedoleron** [comment]
Thanks, @miltonmc! I've made the changes.

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
Hi @miltonmc, please let me know if now it's ok to be forwarded. Thanks!

**Review by miltonmc — APPROVED** [review]
LGTM!

**Comment by miltonmc** [comment]
ci:forward

**Pull request closed without merging**